### PR TITLE
feat: Add local source context via devserver

### PR DESCRIPTION
**Pull request by HazAT**

### CODE REVIEW

File: src/js/integrations/debugsymbolicator.ts [hunk 1]
```diff
@@ -121,25 +118,47 @@ export class DebugSymbolicator implements Integration {
...
+            !frame.file.includes("node_modules") &&
+            !frame.file.includes("native code");
+
+          const newFrame: StackFrame = {
+            colno: frame.column,
+            filename: frame.file,
+            function: frame.methodName,
+            in_app: inApp,
+            lineno: inApp ? frame.lineNumber : undefined, // :HACK
+            platform: inApp ? "javascript" : "node" // :HACK
```
Comment (lobsterkatie) [inline src/js/integrations/debugsymbolicator.ts]: Given that we're now filtering out both node modules _and_ native code, would it be better to make the default `undefined`? Or to test for `frame.file.includes("native code")`?

File: src/js/integrations/debugsymbolicator.ts [hunk 2]
```diff
@@ -158,4 +177,34 @@ export class DebugSymbolicator implements Integration {
...
+        method: "GET"
+      }
+    );
+
+    const content = await response.text();
+    const lines = content.split("\n");
+
+    const line = Math.max(frame.lineno! - 1, 0);
+    frame.pre_context = lines.slice(line - 6, line);
+    frame.post_context = lines.slice(line, line + 5);
```
Comment (lobsterkatie) [inline src/js/integrations/debugsymbolicator.ts]: Maybe I'm reading this wrong, but is there a reason you're grabbing six lines of pre-context but only five lines of post-context? Also, shouldn't the post-context start on the line _after_ the line in question?

**Comment by HazAT** [comment]
I've refactored the code in our code packages since we had this functionality already.
I've also added tests:
https://github.com/getsentry/sentry-javascript/pull/2376

File: src/js/integrations/debugsymbolicator.ts (hunk 2 above)
Comment (HazAT) [reply src/js/integrations/debugsymbolicator.ts]: https://github.com/getsentry/sentry-react-native/pull/741#issuecomment-573024666

Comment (lobsterkatie) [reply src/js/integrations/debugsymbolicator.ts]: 👍🏻

**Pull request merged**